### ISSUE TITLE
Problem with fasta header ???

### ISSUE THREAD

**Issue by davidvilanova** [torognes/swarm#116]
Hi,
While using swarm on dereplicated seqs using vsearch i got the following error

**Vsearch output**
Looks fine
```Dereplicating file ../files/merged_fasta/ALL_combined.fa.gz 100%
127085163 nt in 302191 seqs, min 260, max 446, avg 421
Sorting 100%
74094 unique sequences, avg cluster 4.1, median 1, max 9503
Writing output file 100%
Writing uc file, first part 100%
Writing uc file, second part 100%

```
**swarm output**

`swarm  -f -t 3 -d 1 -z  $outdir/derep.fa -s $outdir/amplicons.stats -u $outdir/ucfinal.txt -w $outdir/OTUs_temp.fa -l $outdir/logswarm > $outdir/amplicons.swarms
`
```
Error: Abundance annotations not found for 74094 sequences, starting on line 1.
>ech97_125
Fasta headers must end with abundance annotations (_INT or ;size=INT).
The -z option must be used if the abundance annotation is in the latter format.
Abundance annotations can be produced by dereplicating the sequences.
The header is defined as the string comprised between the ">" symbol
and the first space or the end of the line, whichever comes first.

```

**The dereplicated file (derep.fa).** 
This is the first sequence of the file
```
>ech97_125 M02944:264:000000000-BVKB2:1:1102:7409:22406;size=9503
TAGGGAATCTTCCGCAATGGACGAAAGTCTGACGGAGCAACGCCGCGTGAACGATGAAGGCCTTCGGGTCGTAAAGTTCTGTTGTTAGGGAAGAACAAGTACCGTTCAAATAGGGCGGTACCTTGACGGTACCTAACCAGAAAGCCACGGCTAACTACGTGCCAGCA
GCCGCGGTAATACGTAGGTGGCAAGCGTTGTCCGGAATTATTGGGCGTAAAGCGCGCGCAGGCGGTCTCTTAAGTCTGATGTGAAATCTCGCGGCTCAACCGCGAGCGGCCATTGGAAACTGGGAGGCTTGAGTGCAGAAGAGGAGAGTGGAATTCCATGTGTAGCG
GTGAAATGCGTAGATATATGGAGGAACACCAGTGGCGAAGGCGACTCTCTGGTCTGTAACTGACGCTGAGGCGCGAAAGCGTGGGGAGCAAACA
```

Using vsearch v2.8.5_linux_x86_64 and Swarm 2.2.2 [Jun 21 2018 17:48:50] on linux.

**Comment by torognes** [comment]
Thanks for reporting this issue.

Could you please provide the command line that you used to dereplicate the sequences with vsearch as well?

The problem seems to be caused by the space in the header line after "ech97_125". Swarm will interpret this as the end of the identifier. It will not find an abundance value in the expected format (`;size=`) because the `-z` option was given and therefore complain. The abundance of 9502 computed by vsearch is ignored.

Usually vsearch will drop the part of the header after the space, but I think it may include it when converting directly from FASTQ.

**Comment by davidvilanova** [comment]
Oh i see,
Here is the command line:

`vsearch --derep_fulllength Sequences.fa.gz --output $outdir/derep.fa --uc=$outdir/derepuc.uc --log=$outdir/log.derep --minuniquesize 1  --fasta_width 0 --sizeout 
`
The input file is a fasta file, containing the space. I can remove everything after the space and keep the ;size=INT

For intance if it change the header to `>ech97_125;size=9503` it should work ?

**Comment by davidvilanova** [comment]
I´ve used awk to clean headers on derep.fa
`awk -F'[ :;]' '{print $1,$NF}' OFS=";" $outdir/derep.fa `

Probably should have cleaned headers early on....

**Comment by torognes** [comment]
Thanks.

>For intance if it change the header to >ech97_125;size=9503 it should work ?

Yes, that should work.

Unfortunately, there seems to be an issue with the `derep_fulllength` command in VSEARCH. It fails to remove everything in the header starting with the space. That's a bug. Sorry for that. I'll fix it.

**Comment by frederic-mahe** [comment]
> I´ve used awk to clean headers on derep.fa
> `awk -F'[ :;]' '{print $1,$NF}' OFS=";" $outdir/derep.fa`

or `sed -i '/^>/ s/ .*;size=/;size=/' file.fas`

Thanks for reporting that issue @davidvilanova

**Comment by frederic-mahe** [comment]
The issue in vsearch: https://github.com/torognes/vsearch/issues/338

**Comment by davidvilanova** [comment]
Thanks guys for quick response !!!

**Comment by torognes** [comment]
The bug is fixed in vsearch 2.8.6, just released.
